### PR TITLE
T#10/error reporting

### PR DESCRIPTION
**Pull request by sudo-may**

### CODE REVIEW

File: backend/src/main/groovy/skills/stress/errors/SeenError.groovy [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.concurrent.ConcurrentLinkedQueue
+
+class SeenError {
+    String id = UUID.randomUUID().toString()
+    Date lastSeen = new Date()
+    String fullError
+    String msg
+    int numOccur = 1
+    String httpStatus
```
Comment (mosefer) [inline backend/src/main/groovy/skills/stress/errors/SeenError.groovy:28]: there are several unused attributes in SeenError, are those no longer used or intended for future expansion?

File: backend/src/main/groovy/skills/stress/errors/SeenError.groovy [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    String serverBody
```
Comment (mosefer) [inline backend/src/main/groovy/skills/stress/errors/SeenError.groovy:29]: there are several unused attributes in SeenError, are those no longer used or intended for future expansion?

File: backend/src/main/groovy/skills/stress/errors/ErrorTracker.groovy [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    int minDistance = 10
+    int numErrorsToKeep = 20
+
+    private final Queue<SeenError> errors = new ConcurrentLinkedQueue()
+
+    private boolean isUserNotFound(SeenError seenError){
+        return seenError.serverBody.contains("\"errorCode\":\"UserNotFound\"")
+    }
+
+    synchronized void track(String statusCode, String response, String body) {
```
[the remark this answers is not in the document]
Comment (sudo-may) [reply backend/src/main/groovy/skills/stress/errors/ErrorTracker.groovy]: good idea, just pushed!

File: backend/src/main/groovy/skills/stress/errors/ErrorTracker.groovy [hunk 4]
```diff
@@ -30,20 +33,41 @@ class ErrorTracker {
     int numErrorsToKeep = 20
 
     private final Queue<SeenError> errors = new ConcurrentLinkedQueue()
+    private final SynchronousQueue<SeenError> processIncomingErrors = new SynchronousQueue<>();
```
Comment (mosefer) [inline backend/src/main/groovy/skills/stress/errors/ErrorTracker.groovy:36]: synchronousqueue is probably not the right data structure here

**Pull request merged**